### PR TITLE
fix: sliders mouseup is not firing when leaving the window

### PR DESCRIPTION
**Pull request by robarbms**
# Pull Request

## 📖 Description

When leaving the window, the slider will not register a mouseup event.

### 🎫 Issues

https://github.com/microsoft/fast/issues/4825

## 👩‍💻 Reviewer Notes

I'm capturing mouseleave on the window document. I found that mouseout fired while still in the window. Perhaps it's firing for the shadow DOMs mouseout? The window element wouldn't fire on mouseleave so I'm using window.document. I attach it on mousedown and remove it on mouseup per the existing pattern.

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/src/slider/slider.ts
Comment (scomea) [inline packages/web-components/fast-foundation/src/slider/slider.ts:468]: I think we should be making sure these listeners are removed on disconnect.

**Comment by scomea** [comment]
are the horizonal scroll changes related to the slider issue?

File: packages/web-components/fast-foundation/src/slider/slider.ts
Comment (robarbms) [reply packages/web-components/fast-foundation/src/slider/slider.ts:468]: This is weird. It's a different PR that I have out. I'll resubmit a corrected PR.

**Review by nicholasrice — APPROVED** [review]

**Review by nicholasrice — CHANGES_REQUESTED** [review]
I think something got messed up with the commit history on this PR. can you fix that before merging?

**Comment by robarbms** [comment]
> are the horizonal scroll changes related to the slider issue?

Another PR got merged some how. I've undone the errant changes

File: packages/web-components/fast-foundation/src/slider/slider.ts
Comment (robarbms) [reply packages/web-components/fast-foundation/src/slider/slider.ts]: There were a lot of listeners, with listeners attaching other listeners. Rather than risk missing one, or a new one being added and by one of the methods, I created a 'remove' flag for each so that I could use the same methods to add and remove event listeners to ensure parity. It feels pretty heavy handed, but I feel like this way would be less prone to errors.

**Review by chrisdholt — APPROVED** [review]

**Comment by chrisdholt** [comment]
@robarbms FYI - looks like there are a couple of conflicts with the Horizontal Scroll changes going in.

**Comment by robarbms** [comment]
> @robarbms FYI - looks like there are a couple of conflicts with the Horizontal Scroll changes going in.

There was a whitespace conflict 😑

**Comment by nicholasrice** [comment]
It looks like git history got messed up on this, would you mind resolving prior to merge?

File: packages/web-components/fast-foundation/src/slider/slider.ts
Comment (nicholasrice) [inline packages/web-components/fast-foundation/src/slider/slider.ts:469]:
```suggestion
     * @param e - MouseEvent or null. If there is no event object provided it will remove the events handlers
```

**Review by nicholasrice — APPROVED** [review]
approved pending the git commit history being cleaned up.

**Pull request merged**
